src/net/tcp/listener.rs

```diff
@@ -64,10 +64,15 @@ impl TcpListener {
     /// established, the corresponding [`TcpStream`] and the remote peer’s
     /// address will be returned.
     pub async fn accept(&self) -> Result<(TcpStream, SocketAddr)> {
+        enum Retry {
+            Now,
+            Wait,
+        }
+
         loop {
-            let maybe_accept = World::current(|world| {
+            let accept_result = World::current(|world| {
                 let host = world.current_host_mut();
-                let (syn, origin) = host.tcp.accept(self.local_addr)?;
+                let (syn, origin) = host.tcp.accept(self.local_addr).ok_or(Retry::Wait)?;
```

Comment:
Could we return from the `World::current` closure here and avoid needing the enum? accept -> None signals the wait, syn.ack.send() -> err signals the retry, and the Ok path remains the same.

Reply:
Just to check my understanding: You are suggesting that we shorten the `World::current` closure to already end here and then do the `syn.ack.send()` and testing of the result outside of it, where we can `continue` the loop if necessary?

That would work, but in the `Ok` case we're have to enter `World::current` again to get access to the `host` back, unless I'm missing something. Still sounds a bit simpler, I'll give it a try!

Reply:
Pushed a change that gets rid of the enum and breaks out of the loop once we know we don't need to retry anymore. Lmk what you think!